### PR TITLE
test: sp db pkg adds unit test

### PR DESCRIPTION
**Pull request by sysvm**
### Description

1. sp db pkg adds unit test
2. gfspclient pkg adds interface for mock use

### Rationale

Ensure SP code quality.

### Example

N/A

### Changes

Notable changes: 
* N/A

### CODE REVIEW

File: base/gfspapp/app.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ type GfSpBaseApp struct {
 
 	server *grpc.Server
 	client *gfspclient.GfSpClient
+	// client gfspclient.GfSpClientAPI
```
Comment (will-2012) [inline base/gfspapp/app.go:31]: Remove it.

File: base/gfspclient/interface.go [hunk 2]
```diff
@@ -1,5 +1,28 @@
...
+	sptypes "github.com/bnb-chain/greenfield/x/sp/types"
+	storage_types "github.com/bnb-chain/greenfield/x/storage/types"
+	storagetypes "github.com/bnb-chain/greenfield/x/storage/types"
+	virtual_types "github.com/bnb-chain/greenfield/x/virtualgroup/types"
+	virtualgrouptypes "github.com/bnb-chain/greenfield/x/virtualgroup/types"
+)
+
 // GfSpClientAPI for mock use
 //
 //go:generate mockgen -source=./interface.go -destination=./interface_mock.go -package=gfspclient
```
Comment (will-2012) [inline base/gfspclient/interface.go:28]: Add this gen cmd to Makefile.

File: store/sqldb/batch.go [hunk 3]
```diff
@@ -6,6 +6,8 @@ package sqldb
 
 // Batch is a write-only database that commits changes to its host database
 // when Write is called. A batch cannot be used concurrently.
+//
+//go:generate mockgen -source=./batch.go -destination=./batch_mock.go -package=sqldb
```
Comment (will-2012) [inline store/sqldb/batch.go:10]: Ditto.

File: store/sqldb/store.go [hunk 4]
```diff
@@ -152,6 +152,89 @@ func InitDB(config *config.SQLDBConfig) (*gorm.DB, error) {
 	return db, nil
 }
 
+// func migrateSpDBTable(db *gorm.DB) (err error) {
```
Comment (will-2012) [inline store/sqldb/store.go:155]: What is this comment?

**Review by will-2012 — APPROVED** [review]

File: base/gfspapp/app.go (hunk 1 above)
Comment (sysvm) [reply base/gfspapp/app.go:31]: fixed

File: base/gfspclient/interface.go (hunk 2 above)
Comment (sysvm) [reply base/gfspclient/interface.go:28]: I think this should have a talk.

File: store/sqldb/batch.go (hunk 3 above)
Comment (sysvm) [reply store/sqldb/batch.go:10]: ditto

File: store/sqldb/store.go (hunk 4 above)
Comment (sysvm) [reply store/sqldb/store.go:155]: deleted

**Review by joeylichang — APPROVED** [review]

**Pull request merged**